<launch>
   <!-- global PARAMS Change here if needed -->
  <arg name="gui" default="false" />
  <arg name="autostart" default="true"/>
  <arg name="serial_device" default="/dev/ttyUSB0"/>
  <arg name="simulation" default="False"/>
  <arg name="logging_config" default="$(find schunk_svh_driver)/etc/logging.xml" />
  <arg name="use_internal_logging" default="False"/>
  <arg name="controller_config" default="$(find schunk_svh_driver)/etc/controller_user.yaml" />
  <arg name="general_config" default="$(find schunk_svh_driver)/etc/svh.yaml" />


  <!-- BEWARE SET THIS TO TRUE TO USE THE HAND WITHOUT ANY OTHER ROS TOOL -->
  <arg name="standalone" default="false"/>

  <!-- load basic configuration -->
  <rosparam file="$(arg general_config)" command="load" ns="svh_controller"/>
  <!-- Load Yaml Configuration file-->
  <rosparam file="$(find schunk_svh_driver)/etc/controller_default.yaml" command="load" ns="svh_controller"/>
  <!-- Load the User Config file which will overwrite values from default -->
  <rosparam file="$(arg controller_config)" command="load" ns="svh_controller"/>
 
  <!-- Robot description for State publishers -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find schunk_svh_driver)/description/urdf/svh-standalone.urdf.xacro" />
  
  
  
  <!-- leap motion nodes -->
    <node pkg="leap_client" type="leap_client_node" name="leap_client">
        <param name="refresh_rate" type="double" value="20.0"/>
        <param name="hands_topic" type="str" value="hand_info"/>
    </node>
  
  <node pkg="tf" type="static_transform_publisher" name="world_tf" args="0 0 0 0 0 0 world leap_client 100" />
  
  <node pkg="leap_client" type="leap_viz" name="leap_viz">
        <param name="hands_topic" type="str" value="/leap_client/hand_info"/>
    </node>
  
  
  
   <!-- svh control node -->
    <node pkg="svh_control" type="Leap_motion_bridge" name="Leap_motion_bridge">
        <remap from="/Leap_motion_bridge/leap_client" to="/leap_client/hand_info"/>
    </node>
    
    
    <!-- Start RQT by default if not specified otherwise -->
  <node if="$(arg gui)" name="rqt_gui" pkg="rqt_gui" type="rqt_gui" />
  
    
    <!-- Launch actual driver node (if not in simulation)-->
  <node unless="$(arg simulation)" name="svh_controller" pkg="schunk_svh_driver" type="svh_controller" output="screen">
    <param name="serial_device" value="$(arg serial_device)"/>
    <param name="autostart" value="$(arg autostart)" />
    <!-- Params for the ICL (driver internal) logging. General configuration in the given logging.xml, output will be saved to the who given files -->
    <param name="logging_config" value="$(arg logging_config)" />
    <param name="use_internal_logging" value="$(arg use_internal_logging)" />
    <remap from="/svh_controller/channel_targets" to="/Leap_motion_bridge/channel_targets" />
  </node>
  

    
</launch>
